### PR TITLE
DBZ-4597 Add Oracle FAQ entry about "Logs don't contain offset"

### PR DESCRIPTION
**Pull request by Naros**
https://issues.redhat.com/browse/DBZ-4597

### CODE REVIEW

File: documentation/modules/ROOT/pages/connectors/oracle.adoc [hunk 1]
```diff
@@ -3505,6 +3505,31 @@ In these abnormal situations, {prodname}, like Kafka, provides _at least once_ d
...
 The rest of this section describes how {prodname} handles various kinds of faults and problems.
 
+[id="oracle-logs-do-not-contain-offset-perform-new-snapshot"]
+=== Logs do not contain offset, perform a new snapshot
+
+When the {prodname} Oracle connector restarts, you may find that the connector reports the following error:
+
+[source]
+----
+Online REDO LOG files or archive logs do not contain the offset scn xxxxxxx. Please perform a new snapshot.
```
Comment (gunnarmorling) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:3515]: For the MySQL connector we have the [`when_needed`](https://debezium.io/documentation/reference/stable/connectors/mysql.html#mysql-property-snapshot-mode) snapshot mode; looks like that'd be a worthwhile addition for the Oracle connector, too? If you agree, @Naros, can you log an issue for this in Jira?

**Review by gunnarmorling — APPROVED** [review]
LGTM, one question inline. Awaiting @roldanbob to take a look as well. Thanks!

File: documentation/modules/ROOT/pages/connectors/oracle.adoc (hunk 1 above)
Comment (Naros) [reply documentation/modules/ROOT/pages/connectors/oracle.adoc:3515]: Done, logged [DBZ-4612](https://issues.redhat.com/browse/DBZ-4612).

**Comment by roldanbob** [comment]
> LGTM, one question inline. Awaiting @roldanbob to take a look as well. Thanks!

Hey, sorry this got lost in the recent barrage of comms. I'll take a look shortly, @Naros!

File: documentation/modules/ROOT/pages/connectors/oracle.adoc [hunk 2]
```diff
@@ -3505,6 +3505,31 @@ In these abnormal situations, {prodname}, like Kafka, provides _at least once_ d
...
+It's worth noting that you may find a record in the `V$ARCHIVED_LOG` table that has a SCN range that matches the referenced SCN; however, that alone doesn't mean the record is available to be mined.
+The record must also have a filename in the `NAME` column, a value of `NO` in the `DELETED` column, and a value of `A` (available) in the `STATUS` column.
+If any of these criteria do not match, then the record is considered incomplete and cannot be mined.
+
+Archive logs must be retained up-to if not slightly longer than your longest downtime window of the connector.
+
+[NOTE]
+====
+When a record has no value in the `NAME` column, this means the log has been removed the file system.
+In this situation, the record is accompanied with `DELETED` being set to `YES` and the `STATUS` column being set to `D` to represent the log has been deleted.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:3530]:
```suggestion
In this situation, the record is accompanied with `DELETED` being set to `YES` and the `STATUS` column being set to `D` to represent the log has been deleted.
```
In such records, the  value of the `DELETED` field is set to `YES`, and the `STATUS` field is set to `D`  to indicate that the log is deleted.

Comment (roldanbob) [reply documentation/modules/ROOT/pages/connectors/oracle.adoc:3530]: In the preceding, is it the _log_ that is deleted or the _record_? That is, should the sentence be:

In this situation, the record is accompanied with `DELETED` being set to `YES` and the `STATUS` column being set to `D` to indicate the record was removed.

**Review by roldanbob — CHANGES_REQUESTED** [review]

File: documentation/modules/ROOT/pages/connectors/oracle.adoc (hunk 2 above)
Comment (Naros) [reply documentation/modules/ROOT/pages/connectors/oracle.adoc:3530]: No, _log_ is correct in this context.  We basically have a record in the database that previously pointed to the log; however, when the log gets removed, the record is kept but the references to the log on the filesystem are nulled out, i.e. `NAME` is set to `NULL`, `DELETED` is set to `YES`, and `STATUS` is set to `D` to indicate deletion.

**Comment by Naros** [comment]
Hi @roldanbob, I've made your suggestions, could you give this one final look-over today if possible so I can merge it for tomorrow's release?  Thanks!

File: documentation/modules/ROOT/pages/connectors/oracle.adoc [hunk 3]
```diff
@@ -3505,6 +3505,32 @@ In these abnormal situations, {prodname}, like Kafka, provides _at least once_ d
 
 The rest of this section describes how {prodname} handles various kinds of faults and problems.
 
+[id="oracle-logs-do-not-contain-offset-perform-new-snapshot"]
+=== Logs do not contain offset, perform a new snapshot
+
+When the {prodname} Oracle connector restarts, you may find that the connector reports the following error:
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:3511]:
```suggestion
In some cases, after the {prodname} Oracle connector restarts, it reports the following error:
```

File: documentation/modules/ROOT/pages/connectors/oracle.adoc [hunk 4]
```diff
@@ -3505,6 +3505,32 @@ In these abnormal situations, {prodname}, like Kafka, provides _at least once_ d
...
+----
+Online REDO LOG files or archive logs do not contain the offset scn xxxxxxx. Please perform a new snapshot.
+----
+
+After the connector examines the redo and archive logs, if it cannot find the SCN that is recorded in the connector offsets, it returns the preceding error.
+Because the connector uses the SCN to determine where to resume processing, if the expected SCN if not found, a new snapshot must be completed.
+
+You might find that the `V$ARCHIVED_LOG` table contains a record with an SCN that matches the expected range.
+However, the record might not be available for mining.
+The record must also have a filename in the `NAME` column, a value of `NO` in the `DELETED` column, and a value of `A` (available) in the `STATUS` column.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:3523]:
```suggestion
To be available for mining, a record must include a filename in the `NAME` column, a value of `NO` in the `DELETED` column, and a value of `A` (available) in the `STATUS` column.
```

File: documentation/modules/ROOT/pages/connectors/oracle.adoc [hunk 5]
```diff
@@ -3505,6 +3505,32 @@ In these abnormal situations, {prodname}, like Kafka, provides _at least once_ d
...
+If any of these criteria do not match, then the record is considered incomplete and cannot be mined.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:3524]:
```suggestion
If a record does not match any of these criteria, it is considered incomplete and cannot be mined.
```

**Review by roldanbob — CHANGES_REQUESTED** [review]
A few tweaks.

**Comment by roldanbob** [comment]
LGTM.

**Comment by gunnarmorling** [comment]
@Naros, could you merge it yourself? This seems to have triggered way to many build checks though?

**Comment by Naros** [comment]
> This seems to have triggered way to many build checks though?

Ya I plan to merge this after i resolve the build triggers.

**Comment by Naros** [comment]
Seems fixed now.  I guess there was an issue with the fork head reference with the new workflows and some changes I made on main that somehow introduced a conflict.  I'll keep an eye out and if you see anything else that's weird in this interim let me know.  A rebase seemed to have fixed it.

**Pull request merged**